dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java

```diff
@@ -316,75 +319,110 @@ private void loadChoiceAuthorityConfigurations() {
-                                        fields = authorityName2definitions.get(submissionName);
-                                    } else {
-                                        fields = new ArrayList<String>();
+                            // do we have an authority?
+                            if (StringUtils.isNotBlank(authorityName)) {
+                                String fieldKey = makeFieldKey(dcinput.getSchema(), dcinput.getElement(),
+                                                               dcinput.getQualifier());
+                                ChoiceAuthority ca = controller.get(authorityName);
+                                if (ca == null) {
+                                    ca = (ChoiceAuthority) pluginService
+                                        .getNamedPlugin(ChoiceAuthority.class, authorityName);
+                                    if (ca == null) {
+                                        throw new IllegalStateException("Invalid configuration for " + fieldKey
+                                                + " in submission definition " + submissionName
+                                                + ", form definition " + dcinputSet.getFormName()
+                                                + " no named plugin found: " + authorityName);
                                     }
-                                    fields.add(fieldKey);
-                                    authorityName2definitions.put(submissionName, fields);
-                                    authoritiesFormDefinitions.put(authorityName, authorityName2definitions);
                                 }
+
+                                addAuthorityToFormCacheMap(submissionName, fieldKey, ca);
+                                addFormDetailsToAuthorityCacheMap(submissionName, authorityName, fieldKey);
                             }
                         }
                     }
                 }
             }
         } catch (DCInputsReaderException e) {
             // the system is in an illegal state as the submission definition is not valid
-            throw new IllegalStateException(e);
+            throw new IllegalStateException("Error reading the item submission configuration: " + e.getMessage(),
+                    e);
+        }
+    }
+
+    /**
+     * Add the form/field to the cache map keeping track of which form/field are
+     * associated with the specific authority name
+     * 
+     * @param submissionName the form definition name
+     * @param authorityName  the name of the authority plugin
+     * @param fieldKey       the field key that use the authority
+     */
+    private void addFormDetailsToAuthorityCacheMap(String submissionName, String authorityName, String fieldKey) {
+        Map<String, List<String>> authorityName2definitions;
+        if (authoritiesFormDefinitions.containsKey(authorityName)) {
+            authorityName2definitions = authoritiesFormDefinitions.get(authorityName);
```

Comment:
This variable name sounds quite confusing. It seems to be a map from the submission name to the fields (and the authorityName is not the same as the submission name)